packages/tempus/src/Tempus_StepperDIRK_decl.hpp

```diff
@@ -203,25 +207,25 @@ class StepperDIRK : virtual public Tempus::StepperImplicit<Scalar>,
   Teuchos::RCP<Thyra::VectorBase<Scalar> >& getXTilde() {return xTilde_;}
 
   /// Return alpha = d(xDot)/dx.
-  virtual Scalar getAlpha(const Scalar dt) const
+  virtual Scalar getAlpha(const Scalar dt) const override
   {
     const Teuchos::SerialDenseMatrix<int,Scalar> & A=this->tableau_->A();
     return Scalar(1.0)/(dt*A(0,0));  // Getting the first diagonal coeff!
```

Comment:
The case for `A(i,i) = 0` is handled specially in the stepper. But this code is troublesome. Can we add a safeguard?

Reply:
Yeah, getAlpha(dt)  for DIRK is 1/(dt * a_ii), but we only return 1/(dt * a_00).  getAlpha(dt) is a base class function for StepperImplicit, so we need to implement something.  Let's do the following, let's test a_ii to see if it zero and take the first nonzero diagonal.  Additionally let's add a new function getAlpha(dt,i) to return the alpha for a particular stage.  If a_ii is zero, we will return std::inf?  Thoughts?

Reply:
that's a good idea